Plugins/DuplicationPlugin/DuplicateContextMenuItem.cs

```diff
@@ -1,156 +1,616 @@
-    {
-        private Dictionary<string, DuplicateAssetExtension> extensions = new Dictionary<string, DuplicateAssetExtension>();
-
-        public DuplicateContextMenuItem()
-        {
-            foreach (var type in Assembly.GetExecutingAssembly().GetTypes())
-            {
-                if (type.IsSubclassOf(typeof(DuplicateAssetExtension)))
-                {
-                    var extension = (DuplicateAssetExtension)Activator.CreateInstance(type);
-                    extensions.Add(extension.AssetType, extension);
-                }
-            }
-            extensions.Add("null", new DuplicateAssetExtension());
-        }
-
-        public override string ContextItemName => "Duplicate";
-
-        public override RelayCommand ContextItemClicked => new RelayCommand((o) =>
-        {
-            EbxAssetEntry entry = App.SelectedAsset as EbxAssetEntry;
-            EbxAsset asset = App.AssetManager.GetEbx(entry);
-
-            DuplicateAssetWindow win = new DuplicateAssetWindow(entry);
-            if (win.ShowDialog() == false)
-                return;
-
-            string newName = win.SelectedPath + "/" + win.SelectedName;
-            newName = newName.Trim('/');
-
-            Type newType = win.SelectedType;
-            FrostyTaskWindow.Show("Duplicating asset", "", (task) =>
-            {
-                try
-                {
-                    string key = entry.Type;
-                    if (!extensions.ContainsKey(entry.Type))
-                        key = "null";
-                    extensions[key].DuplicateAsset(entry, newName, newType != null, newType);
-                }
-                catch (Exception e)
-                {
-                    App.Logger.Log($"Failed to duplicate {entry.Name}");
-                }
-            });
-
-            App.EditorWindow.DataExplorer.RefreshAll();
-        });
-    }
-}
+﻿using AtlasTexturePlugin;
+using DuplicationPlugin.Windows;
+using Frosty.Core;
+using Frosty.Core.Viewport;
+using Frosty.Core.Windows;
+using Frosty.Hash;
+using FrostySdk;
+using FrostySdk.Ebx;
+using FrostySdk.IO;
+using FrostySdk.Managers;
+using FrostySdk.Resources;
+using MeshSetPlugin.Resources;
+using SvgImagePlugin;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DuplicationPlugin
+{
+    public class DuplicationTool
+    {
+        #region --Extensions--
+
+        public class SvgImageExtension : DuplicateAssetExtension
+        {
+            public override string AssetType => "SvgImage";
+
+            public override EbxAssetEntry DuplicateAsset(EbxAssetEntry entry, string newName, bool createNew, Type newType)
+            {
+                EbxAssetEntry refEntry = base.DuplicateAsset(entry, newName, createNew, newType);
+                if (refEntry == null)

```

Comment:
this check seems to be not necessary, since DuplicateAsset will never return null